abigen mistakenly uses ethers_core instead of ethers

If you have ethers-core as a dependency and ethers w abigen feature as a dev-dependency, abigen thinks it should use the `ethers_x` packages instead of `ethers::`

```
error[E0433]: failed to resolve: use of undeclared crate or module `ethers_contract`
  --> tests/solidity.rs:74:1
   |
74 | abigen!(Groth16Verifier, "./tests/verifier_abi.json");
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `ethers_contract`
```

--- a/ethers-core/src/macros/ethers_crate.rs
+++ b/ethers-core/src/macros/ethers_crate.rs
@@ -75,12 +75,6 @@ pub fn determine_ethers_crates() -> (&'static str, &'static str, &'static str) {
                     return sub_crates
                 }
 
-                // Note(mattsse): somehow rustanalyzer can end up expanding `abigen!` from inside the `ethers-middleware` crate: https://github.com/rust-analyzer/rust-analyzer/issues/11518
-                // so we need to bypass detection
-                if pkg.name == "ethers-middleware" || pkg.name == "ethers" {
-                    return None
-                }
-
                 let mut has_ethers_core = false;
                 let mut has_ethers_contract = false;
                 let mut has_ethers_providers = false;

